cardano-node/src/Cardano/Tracing/Startup.hs

```diff
@@ -116,20 +116,37 @@ instance ( Show (BlockNodeToNodeVersion blk)
          , Show (BlockNodeToClientVersion blk)
          )
         => ToObject (StartupTrace blk) where
-    toObject _verb (StartupInfo addresses
-                                 localSocket
-                                 supportedNodeToNodeVersions
-                                 supportedNodeToClientVersions)
-      = mkObject
-        [ "nodeAddresses" .= Aeson.toJSON (map ppN2NSocketInfo addresses)
-        , "localSocket" .= case localSocket of
-              Nothing -> Aeson.Null
-              Just a  -> String (Text.pack . ppN2CSocketInfo $ a)
-        , "nodeToNodeVersions" .=
-            Aeson.toJSON (map show . Map.assocs $ supportedNodeToNodeVersions)
-        , "nodeToClientVersions" .=
-            Aeson.toJSON (map show . Map.assocs $ supportedNodeToClientVersions)
-        ]
+    toObject verb (StartupInfo addresses
+                                localSocket
+                                supportedNodeToNodeVersions
+                                supportedNodeToClientVersions)
+      = case verb of
+          MaximalVerbosity ->
+            mkObject
+              [ "nodeAddresses" .= Aeson.toJSON (map ppN2NSocketInfo addresses)
+              , "localSocket" .= case localSocket of
+                    Nothing -> Aeson.Null
+                    Just a  -> String (Text.pack . ppN2CSocketInfo $ a)
+              , "nodeToNodeVersions" .=
```

Comment:
Some repetition here. Why not:
```
       [ "nodeAddresses" .= Aeson.toJSON (map ppN2NSocketInfo addresses)
       , "localSocket" .= case localSocket of
                    Nothing -> Aeson.Null
                    Just a  -> String (Text.pack . ppN2CSocketInfo $ a)
       ]  ++ case verb of 
                  MaximalVerbosity -> [ ... ] 
                  _ -> [ ... ]
```

Reply:
Because it produces a different object with different keys and the duplication is minimal.  But if you want, I can change this.

Reply:
fixed